File: src/windows/AmbientContextMcp.Core.Tests/TransientStateSettingsTests.cs
using AmbientContextMcp.Core.Settings;
using Xunit;

namespace AmbientContextMcp.Core.Tests;

public class TransientStateSettingsTests : IDisposable
{
    private readonly string _tempDir;
    private readonly string _settingsPath;

    public TransientStateSettingsTests()
    {
        _tempDir = Path.Combine(Path.GetTempPath(), "ambient-context-mcp-test", Guid.NewGuid().ToString());
        Directory.CreateDirectory(_tempDir);
        _settingsPath = Path.Combine(_tempDir, "settings.json");
    }

    public void Dispose()
    {
        try
        {
            Directory.Delete(_tempDir, recursive: true);
        }
        catch
        {
            // テストの後片付けで失敗してもアサーション結果には影響させない
        }
    }

    [Fact]
    public void Save_then_load_round_trips_LastActivityDate()
    {
        var store = new JsonFileSettingsStore(_settingsPath);
        var today = new DateOnly(2026, 5, 17);

        store.SaveTransientStateSettings(new TransientStateSettings { LastActivityDate = today });

        var reopened = new JsonFileSettingsStore(_settingsPath);
        var loaded = reopened.LoadTransientStateSettings();

        Assert.Equal(today, loaded.LastActivityDate);
    }

    [Fact]
    public void Load_returns_default_when_section_absent()
    {
        var store = new JsonFileSettingsStore(_settingsPath);
        // 別セクションだけ保存し、TransientState は触らない
        store.SaveUiSettings(new UiSettings());

        var loaded = store.LoadTransientStateSettings();

        Assert.Null(loaded.LastActivityDate);
    }

    [Fact]
    public void Save_does_not_clobber_other_sections()
    {
        var store = new JsonFileSettingsStore(_settingsPath);
        var customWindow = new SettingsWindowStatus { Left = 123, Top = 456, Width = 789, Height = 321 };
        store.SaveSettingsWindowStatus(customWindow);

        store.SaveTransientStateSettings(new TransientStateSettings { LastActivityDate = new DateOnly(2026, 5, 17) });

        var reopened = new JsonFileSettingsStore(_settingsPath);
        var windowAfter = reopened.LoadSettingsWindowStatus();

        Assert.NotNull(windowAfter);
        Assert.Equal(123, windowAfter!.Left);
        Assert.Equal(456, windowAfter.Top);
    }
}

File: src/macos/Tests/AmbientContextCoreTests/TransientStateSettingsTests.swift
import Foundation
import Testing
@testable import AmbientContextCore

@Suite("TransientStateSettings")
struct TransientStateSettingsTests {
    private let temp = TempDirectory()

    private var settingsPath: String { temp.file("settings.json") }

    @Test("Save_then_load_round_trips_LastActivityDate")
    func saveThenLoadRoundTripsLastActivityDate() {
        let store = JsonFileSettingsStore(path: settingsPath)
        let today = DateOnly(2026, 5, 17)

        store.saveTransientStateSettings(TransientStateSettings(lastActivityDate: today))

        let reopened = JsonFileSettingsStore(path: settingsPath)
        let loaded = reopened.loadTransientStateSettings()

        #expect(loaded.lastActivityDate == today)
    }

    @Test("Load_returns_default_when_section_absent")
    func loadReturnsDefaultWhenSectionAbsent() {
        let store = JsonFileSettingsStore(path: settingsPath)
        // 別セクションだけ保存し、transientState は触らない
        store.saveUiSettings(UiSettings())

        let loaded = store.loadTransientStateSettings()

        #expect(loaded.lastActivityDate == nil)
    }

    @Test("Save_does_not_clobber_other_sections")
    func saveDoesNotClobberOtherSections() throws {
        let store = JsonFileSettingsStore(path: settingsPath)
        store.saveSettingsWindowStatus(
            SettingsWindowStatus(left: 123, top: 456, width: 789, height: 321))

        store.saveTransientStateSettings(
            TransientStateSettings(lastActivityDate: DateOnly(2026, 5, 17)))

        let reopened = JsonFileSettingsStore(path: settingsPath)
        let windowAfter = try #require(reopened.loadSettingsWindowStatus())

        #expect(windowAfter.left == 123)
        #expect(windowAfter.top == 456)
    }
}
